File: 2078.two-furthest-houses-with-different-colors/solution.kt
class Solution {
    fun maxDistance(colors: IntArray): Int {
        var res = 0
        val cl = colors[0]
        val cr = colors[colors.lastIndex]

        colors.asIterable().forEachIndexed { i, c ->
            if (c != cl) {
                res = maxOf(res, i)
            }
            if (c != cr) {
                res = maxOf(res, colors.lastIndex - i)
            }
        }

        return res
    }
}

File: 2078.two-furthest-houses-with-different-colors/solution.go
func maxDistance(colors []int) int {
    res := 0
    cl, cr := colors[0], colors[len(colors) - 1]
    for i := range colors {
        if colors[i] != cl {
            res = max(res, i)
        }
        if colors[i] != cr {
            res = max(res, len(colors) -1 - i)
        }
    }
    return res
}